timemachine/fe/mcgregor.py

```diff
@@ -367,43 +421,24 @@ def recursion(
     if num_edges < threshold:
         return
 
-    if connected_core:
-        # process g1 using atom_map_1_to_2_information
-        g1_node_states = [NODE_STATE_VISITED_DEMAPPED] * g1.n_vertices
-        g1_source = None
-        g1_mapped_count = 0
-        for a1, a2 in enumerate(atom_map_1_to_2):
-            if a1 < layer:
-                # visited nodes
-                if a2 != UNMAPPED:
-                    g1_node_states[a1] = NODE_STATE_VISITED_MAPPED
-                    g1_source = a1
-                    g1_mapped_count += 1
-            else:
-                g1_node_states[a1] = NODE_STATE_UNVISITED
-
-        if g1_source and g1.mapping_is_disconnected(g1_source, g1_node_states, g1_mapped_count):
-            return
+    if max_connected_components is not None or min_connected_component_size > 1:
```

Comment:
It looks like if `min_connected_component_size == 0 ` then `max_connected_components` is ignored. Doesn't look like this case is tested in the code. Should it raise an exception in this case?

Reply:
Sorry, not following... If `min_connected_component_size == 0`, doesn't the condition reduce to `max_connected_components is not None`?

Reply:
Sorry, misunderstood the code.

But in the case of `max_connected_components is not None` and `min_connected_component_size = 0`, should that be allowable? If so would be good to test that case

Reply:
I think it should be allowable -- `min_connected_component_size = 0` is just a no-op filter (same as the default value of 1, or any negative number), so I think this is already covered by `test_max_connected_components`. But this raises the point that we have a test with both filters active at the same time (`max_connected_components is not None and min_connected_component_size > 1`); will consider adding that.

Reply:
If it is allowable, would be good to test for it explicitly then.

Reply:
Added tests for both filters active at the same time in https://github.com/proteneer/timemachine/pull/1349/commits/471ceb3eaf5ae1356b7f21f922617fd9eb77152f, and for `min_connected_component_size=0` in https://github.com/proteneer/timemachine/pull/1349/commits/c0fe6a65f8a9e1285813f1dcd6bf2567e258ba3c